nvm.sh

```diff
@@ -217,6 +217,28 @@ nvm_rc_version() {
   nvm_echo "Found '${NVMRC_PATH}' with version <${NVM_RC_VERSION}>"
 }
 
+nvm_find_package_json() {
+  local dir
+  dir="$(nvm_find_up 'package.json')"
+  if [ -e "${dir}/package.json" ]; then
+    nvm_echo "${dir}/package.json"
+  fi
+}
+
+nvm_package_json_semver_expression() {
+  export PKG_JSON_SEMVER_EXPRESSION=''
+  local PKG_JSON_PATH
+  PKG_JSON_PATH="$(nvm_find_package_json)"
+  if [ ! -e "${PKG_JSON_PATH}" ]; then
+    nvm_err "No package.json file found"
+    return 1
+  fi
+  # TODO handle error case
+  # TODO implement without node?
```

Comment:
Since node is out of the question, is it worth exploring the idea of using regex to read the node version from package.json? The following command looks like it would work?
```
PKG_JSON_SEMVER_EXPRESSION=$(cat package.json \
    | tr '\n' ' ' \
    | grep -o '"engines":\s*{\(\s*"\(node\|npm\)":\s*\"[0-9.<>=*~^xXv\s]\+",\?\s*\)\+}' \
    | grep -o '"node":\s*\"[0-9.<>*~^xXv\s]\+"' \
    | awk -F: '{ print $2 }' \
    | sed 's/[",]//g')
```